### PR TITLE
[core] Modularize batchnorm.

### PR DESCRIPTION
**Pull request by lina128**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/2996)

### CODE REVIEW

**Comment by lina128** [comment]
> Great work! LGTM with three very tiny comments.
> 
> > Reviewed 13 of 13 files at r1.
> > _[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/2996#-:-M3l_-69DnDiqdhPGlk1:b-klenck)_ status: :shipit: complete! 1 of 1 approvals obtained (waiting on @annxingyuan, @dsmilkov, @lina128, and @tafsiri)
> 
> _[tfjs-core/src/kernel_names.ts, line 24 at r1](https://reviewable.io/reviews/tensorflow/tfjs/2996#-M3lYc7eCeBqU7RLW5zI:-M3lYc7eCeBqU7RLW5zJ:b-ggr3e2) ([raw file](https://github.com/tensorflow/tfjs/blob/b285f108256ad00fc1bd76e535c68db488a98b2c/tfjs-core/src/kernel_names.ts#L24)):_
> 
> > ```
> > import {PixelData} from './types';
> > 
> > export const BatchNormalization = 'BatchNormalization';
> > ```
> 
> I believe your observation about this mapping onto FusedBatchNorm is correct so let's use that as the kernel name. In the converter all versions of FusedBatchNorm mapped onto this function https://cs.opensource.google/tensorflow/tfjs/+/master:tfjs-converter/src/operations/executors/normalization_executor.ts;l=31?q=batchnorm&ss=tensorflow%2Ftfjs:tfjs-converter%2F
> 
> _[tfjs-core/src/ops/batchnorm.ts, line 102 at r1](https://reviewable.io/reviews/tensorflow/tfjs/2996#-M3lVLNAA-2RVo1duaCS:-M3lVLNAA-2RVo1duaCT:bv7gw2l) ([raw file](https://github.com/tensorflow/tfjs/blob/b285f108256ad00fc1bd76e535c68db488a98b2c/tfjs-core/src/ops/batchnorm.ts#L102)):_
> 
> > ```
> >           'equal ranks.');
> > 
> >   const x4D: Tensor4D = xAs4D($x);
> > ```
> 
> Let's move this into the forward function.
> 
> _[tfjs-core/src/ops/batchnorm4d.ts, line 93 at r1](https://reviewable.io/reviews/tensorflow/tfjs/2996#-M3lWz5H37yoDZeQhrn8:-M3lWz5H37yoDZeQhrn9:b4a9mb2) ([raw file](https://github.com/tensorflow/tfjs/blob/b285f108256ad00fc1bd76e535c68db488a98b2c/tfjs-core/src/ops/batchnorm4d.ts#L93)):_
> 
> > ```
> > }
> > 
> > export const batchNormalization4d = op({batchNormalization4d_});
> > ```
> 
> Could you add a //todo(yassogba) to remove these later since the batchNormalization* form is already deprecated.

Thank you for the review, Yannick! Done. Also changed all the ops to be called directly instead of chained through Tensor in grad.

**Comment by tafsiri** [comment]
Great! transforming away from chained ops is great to do when we can.
